src/pkynetics/model_fitting_methods/coats_redfern.py

```
@@ -79,9 +86,25 @@
     # Calculate kinetic parameters
     r = 8.314  # Gas constant in J/(mol·K)
     e_a = -slope * r  # Activation energy in J/mol
-    a = np.exp(
-        intercept + np.log(heating_rate / e_a)
-    )  # Pre-exponential factor in min^-1
+
+    # Check if activation energy is physically plausible
+    if e_a <= 0:
+        logger.warning(
```

Ruff (0.8.2)

Undefined name `logger`

(F821)